docs/plugins/pluggable_rule_engine.md

```diff
@@ -32,29 +32,31 @@ Because the rule engine plugins are just one of many iRODS plugin types, the REP
 }
 ```
 
-Within the rule engine plugin framework, there are dynamically created policy enforcement points that are checked before and after every operation.  These are the "_pre", "_post", "_except", and "_finally" PEPs discussed in [Dynamic Policy Enforcement Points](dynamic_policy_enforcement_points.md).
+Within the REPF, there are dynamically created policy enforcement points that are checked before and after every operation.  These are the "_pre", "_post", "_except", and "_finally" PEPs discussed in [Dynamic Policy Enforcement Points](dynamic_policy_enforcement_points.md).
 
 The framework will look for rules that are defined with the same name as the PEP and execute them if found.  A typical `ils` will trigger over 1200 dynamic PEPs on a basic installation.  Nearly all of them will be undefined (there is no rule that matches their name) and so will not run any code.
 
 However, any that *do* match a particular PEP will be executed in the order in which they are loaded.  If there is only one matching rule, then it will fire and its return code will be interpreted by the REPF.  If it fails, then the operation fails as well and an error is returned to the client (or to the log when run by the [delay execution server](#delay-execution).)
 
-If there is more than one matching rule for a particular PEP, the first one loaded will fire first.  If it succeeds, then the others are ignored.  If the first one fails, then the next matching rule is fired.  If it fails, then the framework will continue to "fall through" until there are no more matching rules.  The return code of the last matching rule will be the one that is returned.
+If there is more than one matching rule for a particular PEP, only the first one loaded will fire.  If it succeeds, then the others are ignored.  If the first one fails, then an error is returned to the client and processing stops.
 
-![Rule Engine Plugin Framework Diagram](../images/rule_engine_plugin_framework_diagram.jpg)
+The REPF can be instructed to do different things based on the code returned from a rule. See the following table:
 
-This "fall through" mechanism also applies across rule engine plugins (meaning, across language as well).
+| Symbolic Code Name         | REPF Code | Behavior |
+| -------------------------- | --------- | -------- |
+| RULE_ENGINE_CONTINUE       | 5000000   | Allow the next **rule engine plugin** to process the PEP. |
+| RULE_ENGINE_SKIP_OPERATION | 5001000   | Do not execute the server operation following the **pre** PEP. This can only be returned from a **pre** PEP. |
```

Comment:
'can only be returned'... what happens if it's returned elsewhere?   just treated as an error?  no special handling/meaning/behavior?

Reply:
A warning is logged and the REPF ignores it.

Reply:
please add that clarity/scenario to the behavior.

Reply:
Done.